### PR TITLE
Eq scalar filters

### PR DESCRIPTION
**Pull request by MacondoExpress**
Add the `_EQ` filter and deprecate the implicit equality filters.
Migration of the existing tests will happen in a follow-up PR.

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 44 | 63 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 47 | 73 | 1134 |
|  aggregations.NestedAggregation | 15407 | 15407 | 69 | 71 | 2174 |
|  aggregations.AggregationWithWhere | 10833 | 10833 | 44 | 52 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 20097917 | 20097917 | 2103 | 2270 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 20097917 | 20097917 | 2128 | 2276 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 8603 | 8603 | 32 | 40 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8791 | 8791 | 35 | 42 | 2174 |
|  aggregations.DeeplyNestedCount | 10052335 | 10052335 | 2765 | 2905 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 43 | 49 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 67 | 76 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 132 | 162 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 95 | 60 | 11 |
|  connect.createAndConnect | 6433 | 6433 | 145 | 162 | 3003 |
|  connections.Connection | 12951 | 12951 | 67 | 68 | 2174 |
|  connections.NestedConnection | 37705 | 37705 | 120 | 149 | 4516 |
|  create.SimpleMutation | 7 | 7 | 48 | 56 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 30 | 25 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 637 | 661 | 1040 |
|  delete.NestedDeleteInUpdate | 16844 | 16844 | 166 | 191 | 1183 |
|  2871.NestedRelationshipFilter | 19632 | 19632 | 61 | 65 | 4395 |
|  2925.SingleRelationshipFilter | 5245 | 5245 | 43 | 43 | 1040 |
|  2925.NestedSingleRelationshipFilter | 17641 | 17641 | 79 | 103 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 33 | 41 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 54 | 58 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 20 | 21 | 1040 |
|  query.SimpleQueryWithRelationship | 15031 | 15031 | 37 | 44 | 2174 |
|  query.QueryWhere | 8564 | 8564 | 31 | 36 | 2154 |
|  query.SimpleQueryWithNestedWhere | 8713 | 8713 | 47 | 52 | 2154 |
|  query.Nested | 10084891 | 10084891 | 6909 | 6632 | 2008534 |
|  query.NestedWithFilter | 10064992 | 10064992 | 6477 | 6629 | 2004000 |
|  query.OrFilterOnRelationships | 36682 | 36482 | 142 | 194 | 1939 |
|  query.OrFilterOnRelationshipsAndNested | 26755 | 26913 | 186 | 303 | 1939 |
|  query.QueryWithNestedIn | 14141 | 14135 | 47 | 73 | 1847 |
|  query.NestedConnectionWhere | 8703 | 8703 | 47 | 68 | 2174 |
|  query.DeeplyNestedConnectionWhere | 8702 | 8702 | 66 | 95 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 17357 | 17357 | 120 | 163 | 1552 |
|  query.NestedWithRelationshipSingleFilters | 3808 | 3808 | 145 | 179 | 1134 |
|  query.Fulltext | 64 | 64 | 37 | 57 | 16 |
|  query.FulltextWithNestedQuery | 516 | 516 | 41 | 48 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 12961 | 12961 | 38 | 46 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 12961 | 12961 | 58 | 67 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 13096 | 13096 | 60 | 61 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 13096 | 13096 | 88 | 98 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 13690 | 13705 | 95 | 116 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 13266 | 13266 | 107 | 139 | 2174 |
|  sorting.SortMultipleTypes | 3436 | 3436 | 77 | 92 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 13321 | 13321 | 94 | 102 | 2174 |
|  sorting.SortOnNestedFields | 12951 | 12951 | 41 | 52 | 2174 |
|  sorting.SortDeeplyNestedFields | 39785 | 39785 | 77 | 119 | 4516 |
|  sorting.ConnectionWithSort | 3271 | 3271 | 75 | 70 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 52 | 64 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 104 | 56 | 35 |
|  unions.NestedUnion | 309975 | 309975 | 294 | 274 | 33033 |
|  unions.NestedUnionWithMissingFields | 283949 | 283949 | 247 | 263 | 33033 |
|  update.NestedUpdate | 14137 | 14137 | 101 | 109 | 2002 |

</details>
        
Old Schema Generation: 27.013s
Schema Generation: 26.976s
Old Subgraph Schema Generation: 28.671s
Subgraph Schema Generation: 29.010s

File: packages/graphql/src/translate/queryAST/factory/parsers/parse-where-field.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export type WhereRegexGroups = {
 };
 
 export const whereRegEx =
-    /(?<prefix>\w*\.)?(?<fieldName>[_A-Za-z]\w*?)(?<isConnection>Connection)?(?<isAggregate>Aggregate)?(?:_(?<operator>NOT|NOT_IN|IN|NOT_INCLUDES|INCLUDES|MATCHES|NOT_CONTAINS|CONTAINS|NOT_STARTS_WITH|STARTS_WITH|NOT_ENDS_WITH|ENDS_WITH|LT|LTE|GT|GTE|DISTANCE|ALL|NONE|SINGLE|SOME))?$/;
+    /(?<prefix>\w*\.)?(?<fieldName>[_A-Za-z]\w*?)(?<isConnection>Connection)?(?<isAggregate>Aggregate)?(?:_(?<operator>NOT|NOT_IN|IN|NOT_INCLUDES|INCLUDES|MATCHES|NOT_CONTAINS|CONTAINS|NOT_STARTS_WITH|STARTS_WITH|NOT_ENDS_WITH|ENDS_WITH|EQ|LT|LTE|GT|GTE|DISTANCE|ALL|NONE|SINGLE|SOME))?$/;
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/factory/parsers/parse-where-field.ts:32]: magical :magic_wand:

**Review by angrykoala — DISMISSED** [review]

File: packages/graphql/src/translate/queryAST/factory/parsers/parse-where-field.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/translate/queryAST/factory/parsers/parse-where-field.ts:32]: 🤣

Comment (darrellwarde) [reply packages/graphql/src/translate/queryAST/factory/parsers/parse-where-field.ts:32]: Ahhhhh this RegEx brings back memories... 🥲

**Pull request merged**